charts/application/templates/_podTemplate.tpl

```diff
@@ -51,7 +51,11 @@ spec:
   initContainers: {{ if not .Values.initContainers -}}[]{{- end -}}
     {{- range $i := .Values.initContainers }}
     - name: {{ $i.name }}
+      {{- if .image.tag }}
       image: "{{ .image.repository }}:{{ .image.tag }}"
+      {{- else }}
+      image: "{{ .image.repository }}@{{ .image.hash }}"
+      {{- end }}
```

Comment:
IMO, this should be reversed
```suggestion
      {{- if .image.hash }}
      image: "{{ .image.repository }}@{{ .image.hash }}"
      {{- else }}
      image: "{{ .image.repository }}:{{ .image.tag }}"
      {{- end }}
```
otherwise one must always unset the tag to be able to used the hash.

Reply:
Also I think it's better to define a variable like `imageTag` in https://github.com/MediaMarktSaturn/helm-charts/blob/main/charts/application/templates/_helpers.tpl
instead of having these if-else repeated everywhere.

Reply:
i tried to define a variable in the template, but i have no experience with this. can you give ne a short example?